File: 백준/Silver/17103. 골드바흐 파티션/골드바흐 파티션.py
import sys

prime = [False, False] + [True] * 999_999
for i in range(2, 1_000_001):
    if prime[i]:
        for j in range(i * 2, 1_000_001, i):
            prime[j] = False
t = int(sys.stdin.readline())
for _ in range(t):
    n = int(sys.stdin.readline())
    cnt = 0
    for i in range(2, n // 2 + 1):
        if prime[i] and prime[n - i]:
            cnt += 1
    print(cnt)

File: 백준/Silver/17103. 골드바흐 파티션/골드바흐 파티션.cc
#include <iostream>
#include <cmath>
using namespace std;

const int RANGE = 1'000'001;
int prime[RANGE];

int main() {
	cin.tie(0)->sync_with_stdio(0);

	fill_n(prime, RANGE, 1);
	prime[0] = prime[1] = 0;
	for (int i = 2; i * i <= RANGE; i++) {
		if (prime[i]) {
			for (int j = i * 2; j <= RANGE; j += i) {
				prime[j] = 0;
			}
		}
	}
	int t;
	cin >> t;
	while (t--) {
		int n;
		cin >> n;
		int cnt = 0;
		for (int i = 2; i <= n / 2; i++) {
			if (prime[i] && prime[n - i]) cnt++;
		}
		cout << cnt << '\n';
	}
}